esphome/components/esp32_spi_led_strip/led_strip.cpp

```diff
@@ -0,0 +1,191 @@
+#ifdef USE_ESP32
+#include "esphome/core/log.h"
+
+#include "led_strip.h"
+
+namespace esphome {
+namespace esp32_spi_led_strip {
+
+static const char *const TAG = "esp32_spi_led_strip";
+
+inline static const char *rgb_order_to_string(RGBOrder order) {
+  switch (order) {
+    case RGBOrder::ORDER_RGB:
+      return "RGB";
+    case RGBOrder::ORDER_RBG:
+      return "RBG";
+    case RGBOrder::ORDER_GRB:
+      return "GRB";
+    case RGBOrder::ORDER_GBR:
+      return "GBR";
+    case RGBOrder::ORDER_BGR:
+      return "BGR";
+    case RGBOrder::ORDER_BRG:
+      return "BRG";
+    default:
+      return "UNKNOWN";
+  }
+}
+
+void LedStripSpi::dump_config() {
+  ESP_LOGCONFIG(TAG, "ESP32 SPI LED Strip:");
+  ESP_LOGCONFIG(TAG, "  Data Pin: GPIO%d", this->data_pin_);
+  ESP_LOGCONFIG(TAG, "  Clock Pin: GPIO%d", this->clock_pin_);
+  ESP_LOGCONFIG(TAG, "  Number of LEDs: %d", this->num_leds_);
+  ESP_LOGCONFIG(TAG, "  RGB Order: %s", rgb_order_to_string(this->rgb_order_));
+  if (this->max_refresh_rate_) {
+    ESP_LOGCONFIG(TAG, "  Max Refresh Rate: %u Hz", this->max_refresh_rate_);
```

Comment:
For IDF 5+ compatibility:
```suggestion
    ESP_LOGCONFIG(TAG, "  Max Refresh Rate: %" PRIu32 " Hz", this->max_refresh_rate_);
```

Reply:
...you'll also need to `#include <cinttypes>`

Reply:
done.

Reply:
do I also change %d -> PRId16 for num_leds_ and %d -> PRId8 for data_pin_ and clock_pin_?

Reply:
No need for those -- this is a change made in IDF 5+ that only affects 32-bit `int` types. 😄